mitxgraders/sampling.py

```diff
@@ -639,7 +639,7 @@ def gen_symbols_samples(symbols, samples, sample_from, functions, suffixes):
     for _ in range(samples):
         # Generate independent samples
         sample_dict = {symbol: sample_from[symbol].gen_sample() for symbol in independent}
-
+        sample_dict.update(constants)
```

Comment:
I think this needs to come before the previous line (well, effectively). Otherwise, you can never override a constant.

Reply:
good catch!

Reply:
That's why we work in pairs!